napari/_app_model/_app.py

```diff
@@ -32,7 +30,7 @@ def __init__(self, app_name=APP_NAME) -> None:
 
     @classmethod
     def get_app_model(cls, app_name: str = APP_NAME) -> NapariApplication:
-        return Application.get_app(app_name) or cls()  # type: ignore[return-value]
+        return cls()  # type: ignore[return-value]
```

Comment:
This is intended to have singleton of NapariApplication. It should be kept. Your code changed the semantic of this method.